index.html

```diff
@@ -231,16 +231,44 @@ <h2>
         Interfacing with a payment handler
       </h2>
       <p>
-        The <dfn>steps to constrain a payment handler</dfn> with
+        The <dfn>steps to filter payment handlers</dfn> with
         <a>BasicCardRequest</a> <var>data</var> are given by the following
-        algorithm. If the end user inputs or selects a <a>card</a> that meets
-        the constraints of <var>data</var>, the algorithm returns a card as a
-        <a>BasicCardResponse</a>.
+        algorithm. If the payee invokes
+        <code>paymentRequest.canMakePayment()</code> or
+        <code>paymentRequest.show()</code>, the algorithm returns
+        <code>true</code> for matching cards. In case of
+        <code>canMakePayment()</code>, the output of this algorithm is returned
+        to the payee. In case of <code>show()</code>, the output of this
+        algorithm determines whether a payment handler is presented to the
+        user.
       </p>
       <ol class="algorithm" data-link-for="BasicCardResponse">
         <li data-link-for="BasicCardRequest">Let <var>requestedNetworks</var>
-        be the result of filtering <var>data</var>["<a>supportedNetworks</a>"]
-        for <a>networks</a> that are <a>known</a> by this payment handler.
+        be <var>data</var>["<a>supportedNetworks</a>"].
+        </li>
+        <li data-link-for="BasicCardRequest">Let <var>requestedTypes</var> be
+        <var>data</var>["<a>supportedTypes</a>"].
+        </li>
+        <li>If the <a>primary account number</a> is not from one of the
+        <a>networks</a> in <var>requestedNetworks</var> and
+        <var>requestedNetworks</var> is not empty, return <code>false</code>.
+        </li>
+        <li>Otherwise, if the <a>primary account number</a> is not from one of
+        the <a>types</a> in <var>requestedTypes</var> and
+        <var>requestedTypes</var> is not empty, return <code>false</code>.
+        </li>
+        <li>Otherwise, return <code>true</code>.
+        </li>
+      </ol>
+      <p>
+        The <dfn>steps to select a payment handler</dfn> are given by the
```

Comment:
These are really the "steps to respond to a payment request". 

In Payment Request API, in [17.4 User accepts the payment request algorithm](https://w3c.github.io/payment-request/#user-accepts-the-payment-request-algorithm)  

We need to change: 

> 8. Set the details attribute value of response to an object containing the payment method specific object that will be used by the merchant to process or validate the transaction. The format of this response will be defined for each payment method.

To:

> 8. Set the details attribute value of response the result of running "steps to respond to a payment request" for the payment handler that is handling this request for payment. That will result in a payment-method specific object that is used by the merchant to process or validate the transaction. The format of this response is defined for each payment method (e.g., see "steps to respond to a payment request in [[!Payment-Method-Basic-Card]]).

Reply:
> (e.g., see "steps to respond to a payment request in [[!Payment-Method-Basic-Card]]).

Again, would really prefer to avoid referencing Basic Card from Payment Request

Reply:
"e.g." means "for example, so should be OK as an example only. What do you think, @adrianhopebailie?